pysheaf.py

```diff
@@ -52,11 +52,49 @@ def isCoface(self,index,orientation=None):
             return (index,orientation) in [(cf.index,cf.orientation) for cf in self.cofaces]
         
 class CellComplex:
-    def __init__(self,cells=[]):
+    def __init__(self,cells=None):
         """Construct a cell complex from its cells"""
-        self.cells=cells
+        if cells == None:
+            self.cells=[]
+        else:
+            self.cells=cells
+
+        # Register cell names with indices into the self.cells list        
+        self.cell_dict={}
+        self.coface_dict={}
+        for i,c in enumerate(self.cells):
+            if c.name == None: # Build names if not present
+                c.name = str(i)
+            self.cell_dict[c.name]=i
+
+    def add_cell(self,cell):
+        """Add a cell to the cell complex"""
+
+        if cell.name == None: # Construct a name if needed
```

Comment:
I would avoid using the len(self.cells) anywhere. How about using an ID counter in the CellComplex constructor initialized at 0 and as cells are added they get an ID. It makes sense that the ID is a property of the Complex it belongs to. That way when we reference a cell's ID or want to add (and eventually delete a cell we have a UID and not a position to reference the cell with. The cell's dictionary should have an ID key. You could still use this ID as a name as you are doing here.